client/state/user-settings/reducer.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+import { merge, omit } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	USER_SETTINGS_UPDATE,
+	USER_SETTINGS_UNSAVED_CLEAR,
+	USER_SETTINGS_UNSAVED_SET,
+	USER_SETTINGS_UNSAVED_REMOVE,
+} from 'state/action-types';
+
+export const settings = ( state = false, action ) => {
+	switch ( action.type ) {
+		case USER_SETTINGS_UPDATE:
+			return merge( {}, state, action.settingValues );
+
+		default:
+			return state;
+	}
+};
```

Comment:
looks like we could simplify this reducer quite a bit…

```js
export const settings = ( state = false, { type, settingValues } ) =>
	USER_SETTINGS_UPDATE === type
		? { ...state, ...settingValues }
		: state;
```

also it seems odd to be storing the default value for _an object's state_ as `false`. might `null` be more appropriate here as the initial state?